src/commands/perform-tasks.ts

```diff
@@ -91,6 +98,38 @@ export class PerformTasksCommand extends BaseCliCommand<IPerformTasksCommandArgs
 
     }
 
+    private skipNonMatchingLeafTasks(tasks: IBuildTask[], taskMatcher: string, tasksPrefix: string): number {
```

Comment:
some of the subtilities might have gone lost on me, but the way this code seems to work is:
- the skip configuration of tasks that were not matched will be set to `true`
- the skip config of tasks that does match will remain true|false|undefined.

i think there are two flaws in here:
- if i want to run a task called "Foo" that is skipped in the tasks file i would expect it to run.
- if i want to run a task called "Foo" that depends on "Bar", the "Bar" task will be skipped, unless Foo is explicitly set to "Skip:false" the Foo task would also get skipped (as it depends on Bar)

i think both can be fixed by explicitly setting the `skip` to either true or false when running with a matcher.

i also see there is something about leaftasks (i didnt fully understand yet), but i believe the following would work:
``` typescript
    private skipNonMatchingLeafTasks(tasks: IBuildTask[], taskMatcher: string, tasksPrefix: string): void {
        for(const task of tasks) {
            const taskFullName = `${tasksPrefix}${task.name}`;

            const skipTask = !minimatch(taskFullName, taskMatcher);
            if(skipTask !== task.skip) {
                ConsoleUtil.LogInfo(`running task ${taskFullName}, as it matched the '${taskMatcher}' pattern`);
                task.skip = skipTask;
            }
        }
    }

```